src/main/kotlin/com/openlattice/rhizome/hazelcast/ChunkedQueueSequence.kt

```diff
@@ -0,0 +1,24 @@
+package com.openlattice.rhizome.hazelcast
+
+import java.lang.IllegalArgumentException
+import java.util.concurrent.BlockingQueue
+
+class ChunkedQueueSequence<T>(private val queue: BlockingQueue<T>, private val chunkSize: Int): Sequence<List<T>> {
+    init {
+        if (chunkSize < 1) {
+            throw IllegalArgumentException("chunkSize")
+        }
+    }
+    override fun iterator(): Iterator<List<T>> {
+        return iterator() {
+            while(true) {
```

Comment:
Is the assumption here that the queue will never be empty and/or any code using this will use `iter.next().isNotEmpty()` rather than `it.hasNext()` ?

Reply:
BlockingQueue has no close semantics, so closing the channel always has to happen in some other way, in or out of band.

In-process queues could put a sentinel object in the queue or use `Thread.interrupt()`.

In Hazelcast, the queue can be destroyed which would throw `DistributedObjectDestroyedException`.

Reply:
I vote for encoding this info in a class description comment